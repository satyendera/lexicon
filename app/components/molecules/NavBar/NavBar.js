/* eslint-disable jsx-a11y/anchor-is-valid */

// @flow
import React from 'react';
import type { Node } from 'react';
import { AnchorVanilla as Anchor } from '@xt-pagesource/atomic-react-pattern-lib';
import Link from 'next/link';
import styles from './NavBar.style';
import withStyles from '../../../lib/withStyles';
import type { Props } from './types';
import { isApplicationLink } from '../../../utils/isApplicationLink';

const NavBar = ({ className, items, children }: Props): Node => (
  <nav className={className}>
    {children || (
      <ul>
        {Array.isArray(items) &&
          items.map(({ label, href }, key) => (
            <li role="none" key={key.toString()}>
              {isApplicationLink(href) ? (
                <Link href={href} role="menuitem">
                  <a>{label}</a>
                </Link>
              ) : (
                <Anchor to={href} role="menuitem">
                  {label}
                </Anchor>
              )}
            </li>
          ))}
      </ul>
    )}
  </nav>
);

NavBar.defaultProps = {
  items: [],
};

export default withStyles(NavBar, styles);

export { NavBar };
