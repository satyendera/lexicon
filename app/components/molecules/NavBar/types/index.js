// @flow
import type { Node } from 'react';

export type NavItem = { label: string, href: string };

export type Props = {
  className: string,
  children?: Node,
  items?: Array<NavItem>,
};
