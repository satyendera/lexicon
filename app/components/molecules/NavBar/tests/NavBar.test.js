import { shallow } from 'enzyme';
// import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { NavBar } from '../NavBar';

const items = [{ href: '/1', label: 'Link 1' }, { href: '/2', label: 'Link 2' }];

describe('<NavBar />', () => {
  test('should render correctly', () => {
    const NavBarComponent = shallow(<NavBar>Test</NavBar>);
    expect(NavBarComponent).toMatchSnapshot();
  });

  test('should render correctly', () => {
    const NavBarComponent = shallow(<NavBar items={items} />);
    expect(NavBarComponent).toMatchSnapshot();
    expect(NavBarComponent.find('li').length).toEqual(2);
  });
});
