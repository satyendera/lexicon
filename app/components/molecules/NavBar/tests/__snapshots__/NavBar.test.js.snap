// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render correctly 1`] = `
<nav>
  Test
</nav>
`;

exports[`<NavBar /> should render correctly 2`] = `
<nav>
  <ul>
    <li
      key="0"
      role="none"
    >
      <Link
        href="/1"
        role="menuitem"
      >
        <a>
          Link 1
        </a>
      </Link>
    </li>
    <li
      key="1"
      role="none"
    >
      <Link
        href="/2"
        role="menuitem"
      >
        <a>
          Link 2
        </a>
      </Link>
    </li>
  </ul>
</nav>
`;
